### PR TITLE
SVGSVGElement's currentView + useCurrentView are non-standard

### PR DESCRIPTION
**Pull request by Ahmad-S792**
<pre>
SVGSVGElement's currentView + useCurrentView are non-standard
<a href="https://bugs.webkit.org/show_bug.cgi?id=226721">https://bugs.webkit.org/show_bug.cgi?id=226721</a>
rdar://problem/79281167

Reviewed by NOBODY (OOPS!).

This patch removes 'currentView' and 'useCurrentView' from
SVGSVGElement.idl file, since they are non-standard.

* Source/WebCore/svg/SVGSVGElement.idl: As above
* LayoutTests/imported/w3c/web-platform-tests/svg/historical-expected.txt: Rebaselined
* LayoutTests/svg/animations/view-dependency-crash-expected.txt: Rebaselined
* LayoutTests/svg/dom/SVGViewSpec-defaults.html: Deleted
* LayoutTests/svg/dom/SVGViewSpec-defaults-expected.txt: Deleted
* LayoutTests/svg/dom/SVGViewSpec-invalid-ref-crash.html: Deleted
* LayoutTests/svg/dom/SVGViewSpec-invalid-ref-crash-expected.txt: Deleted
* LayoutTests/svg/dom/SVGViewSpec.html: Deleted
* LayoutTests/svg/dom/SVGViewSpec-expected.txt: Deleted
* LayoutTests/svg/dom/SVGViewSpec-multiple-views.html: Deleted
* LayoutTests/svg/dom/SVGViewSpec-multiple-views-expected.txt: Deleted
* LayoutTests/svg/dom/viewspec-parser-1-expected.txt: Rebaselined
* LayoutTests/svg/dom/viewspec-parser-2-expected.txt: Rebaselined
* LayoutTests/svg/dom/viewspec-parser-3-expected.txt: Rebaselined
* LayoutTests/svg/dom/viewspec-parser-4-expected.txt: Rebaselined
* LayoutTests/svg/dom/viewspec-parser-5-expected.txt: Rebaselined
* LayoutTests/svg/dom/viewspec-parser-6-expected.txt: Rebaselined
* LayoutTests/svg/dom/viewspec-parser-7-expected.txt: Rebaselined
</pre>
https://github.com/WebKit/WebKit/commit/2e87388142d38d99f7e565229dc41fe6df4a4370

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/16104 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/16421 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/16825 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/17866 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/15113 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/16289 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/19438 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/16528 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/17529 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/16297 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/16762 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/13750 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/18628 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/14005 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/14568 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/21412 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/14994 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/14733 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/17971 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/15325 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/12994 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/14554 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/18923 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/15148 "Built successfully") | | |

### CODE REVIEW

File: Source/WebCore/svg/SVGSVGElement.idl [hunk 1]
```diff
@@ -34,9 +34,6 @@
 
     attribute float currentScale;
     [SameObject] readonly attribute SVGPoint currentTranslate;
-    
-    readonly attribute boolean useCurrentView;
-    readonly attribute SVGViewSpec currentView;
```
Comment (rniwa) [inline Source/WebCore/svg/SVGSVGElement.idl:39]: We should probably also remove the C++ implementations of these properties if we're deleting them.

However, it's hard to gauge the backwards compatibility risk. How do we know it's safe to remove these properties?

Comment (Ahmad-S792) [reply Source/WebCore/svg/SVGSVGElement.idl:39]: These are not part of SVG2 spec and already have been removed from other browsers. I looked into Chromium / Blink patch and patch on the bug itself and both are just touching IDL file.

Comment (nikolaszimmermann) [reply Source/WebCore/svg/SVGSVGElement.idl:39]: It was part of SVG 1.1, but never saw it used in the wild…

Comment (Ahmad-S792) [reply Source/WebCore/svg/SVGSVGElement.idl:39]: Fixed tests and deleted, which are not needed.

**Comment by Ahmad-S792** [comment]
> To be extra cautious, we'd like to place these features behind a preference. You can implement that analogously to [99ff083](https://github.com/WebKit/WebKit/commit/99ff08340f12db8d867a0ed04f176802b7e0a355) and then once we decide from what SDK we want to no longer support these someone from Apple can add the relevant code for that.
> 
> I will be updating #17112 in the same manner (except for the attributes that already return null).

Sure, let me try. If I need help, will disturb you. :-)

**Pull request closed without merging**